File: c++-programming/chapter-007/override.cpp
#include <iostream>

using namespace std;

class Man
{
	public:
		void speak() { cout << "Hello! " << endl; }
		void speak(string msg) { cout << " " << msg << endl; }
};

class Hombre : public Man
{
	public:
		void speak(string msg) { cout << msg << endl; }
};

int main()
{
	Man henry;
	Hombre enrique;

	henry.speak();
	henry.speak("It's a beautiful evening.");

	enrique.speak("Hola!");
	enrique.Man::speak("Es una tarde hermosa.");

	return 0;
}

File: python/chapter-007/override.py
from Man import *
from Hombre import *

guy_1 = Man('Richard')
guy_2 = Hombre('Ricardo')

guy_1.speak('It\'s a beautiful evening.\n')
guy_2.speak('Es una tarde hermosa.\n')

Person.speak(guy_1)
Person.speak(guy_2)
